book/src/css.md

```diff
@@ -0,0 +1,266 @@
+We also set the `background-image` to `none` and add a bit of transition time.
+
+<span class="filename">Filename: listings/css/6/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/6/style.css}}
+```
+
+If we now hover over the button, we see that over the span of one second its background turns yellow and its font turns red.
+After we removed the cursor, the button returns to its original state.
+
+<div style="text-align:center">
+ <video autoplay muted loop>
+  <source src="vid/css_6.webm" type="video/webm">
+Your browser does not support the video tag.
+ </video>
+</div>
+
+## Nodes
+
+In the previous examples a widget always corresponded to a single CSS node.
+This is not always the case.
+For example, [`gtk::ComboBoxText`](../docs/gtk4/struct.ComboBoxText.html) has multiple CSS nodes.
+Let us see how it works.
+
+First, we create a `ComboBoxText`, populate it with the entries "Factory", "Home" and "Subway" and choose "Factory" as the default entry.
+
+
+<span class="filename">Filename: listings/css/7/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/css/7/window/window.ui}}
+```
+
+A `ComboBoxText` contains a label and a small image of an arrow.
+Let us say we want to give the arrow a different color, but leave the label as is.
+How would we do that?
+
+We see the answer by checking the CSS nodes of [`gtk::ComboBox`](https://gtk-rs.org/gtk4-rs/stable/latest/docs/gtk4/struct.ComboBox.html#css-nodes), the more general form of `ComboBoxText`.
+
+```
+combobox
+├── box.linked
+│   ╰── button.combo
+│       ╰── box
+│           ├── cellview
+│           ╰── arrow
+╰── window.popup
+```
+
+We see that the `combobox` node has children, which themselves can have children and attached style classes.
+Now we know that we have to add a CSS rule that applies to the `arrow` node, which is a descendant of `combobox`.
+
+<span class="filename">Filename: listings/css/7/style.css</span>
+
+```css
+{{#rustdoc_include ../listings/css/7/style.css}}
+```
+
+Indeed, we get a `ComboBoxText` with a red arrow.
+
+<div style="text-align:center"><img src="img/css_7.png"/></div>
+
+## Images
+
+Do we really want to settle with an arrow?
+Maybe there are much nicer icons out there you do not even know about yet.
+Fear not, the [Icon Library](https://apps.gnome.org/app/org.gnome.design.IconLibrary/) has you covered there.
+Scroll through the provided icons and choose one from the pre-installed system icons section.
+It's possible to embed icons from other sections too, we'll learn about that in a follow-up chapter.
+
+For now, we will take the icon with the semantic name "format-justify-fill-symbolic".
+In the screenshot below you see that this corresponds to three parallel lines.
+However, on a different system this icon might look different or not exist at all.
+This is why it is typically better to embed the icons you use.
+
+GTK provides its own parameter for loading icons in buttons and expanders: `-gtk-icon-source`.
```

Comment:
I don't think it works in buttons, does it? It works with a few very specific widgets:

- `GtkBuiltinIcon` (GtkCheckButton check etc, and likely this arrow as well)
- `GtkSpinner`
- `GtkTextHandle` (touch text selection)
- `GtkTreeView` somwhewhere - likely the arrow in headers
- `GtkPaned` handle

Maybe a few more, but - it's private and e.g. `GtkImage` doesn't support it. Chances are it will get deprecated before GTK 5.